packages/@react-stately/layout/src/ListLayout.ts

```diff
@@ -253,7 +249,7 @@ export class ListLayout<T> extends Layout<Node<T>, ListLayoutProps> implements K
       y = loader.rect.maxY;
     }
 
-    if (nodes.length === 0) {
+    if (nodes.length === 0 && this.enableEmptyState) {
```

Comment:
In RAC, the loading and empty states are handled in the component rather than in the layout. Ideally this would also the the case in RSP, but I haven't found a good way to implement that yet.